### ISSUE TITLE
fast_open:true in server config causes client phone reboot

### ISSUE THREAD

**Issue by DKingCN** [shadowsocks/shadowsocks-android#1121]
Please read FAQ then answer these questions before submitting your issue. Thanks!

### Environment

* Android version: `7.1.1_r11_angler_factory  &  7.1.1_r16_angler_AOSP`
* Device: `Nexus 6P`
* Shadowsocks version: 
                `SS-Android: v3.3.1 from GPlay`
                `Py-SSServer: v2.9.1,  OS: Kernel 3.13.0-86-generic  Ubuntu 14.04.4 x64`

### Configuration

_Put an `x` inside the [ ] that applies._

* [x] IPv4 server address
* [ ] IPv6 server address
* [x] Client IPv4 availability
* [x] Client IPv6 availability
* Local port: 1080
* Encrypt method: aes-256-cfb
* [ ] One-time authentication
* Route
  * [x] All
  * [ ] Bypass LAN
  * [ ] Bypass China
  * [ ] Bypass LAN & China
  * [ ] GFW List
  * [ ] China List
  * [ ] Custom rules
* [ ] IPv6 route
* [ ] Per-App Proxy
  * [ ] Bypass mode
* [ ] UDP Forwarding
* Plugin configuration (if applicable):
* [ ] Auto Connect
* [x] TCP Fast Open
* [ ] NAT mode

### What did you do?
I did some test and finnal found a simple reason why my phone got Stuck.
Basically my server config file is formed like
{
...
"port_password":{  },
"fast_open":true,
"workers":10
}

With Server.conf "fast_open":true, and client ss-android TFO=3,   Phone instantly stuck then reboot.

Test: Procedure:
Later on I deleted the tfo line in server.conf and restart service, ss-android could work normally with tfo=3.
Then with previous connection kept, I add the fast_open option and restart the service, phone stuck again.

Now we can see that TFO option in both sides will kill each other but I dont know why.
Yet I cant decide is it the kernel problem or ss-android.

**Comment by madeye** [comment]
Disable TFO on your client. It's a bug of Google.

https://github.com/shadowsocks/shadowsocks-android/issues/847

**Comment by DKingCN** [comment]
thanks
